data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder2.java

```diff
@@ -1388,20 +1388,16 @@ private <T> List<T> merge(List<T> left, List<T> right) {
         return associations;
     }
 
-    /**
-     * Quote a column name for the dialect.
-     *
-     * @param persistedName The persisted name.
-     * @return The quoted name
-     */
     @Override
-    protected String quote(String persistedName) {
+    protected String quote(String persistedName, boolean supportsDynamicValues) {
         return switch (dialect) {
             case MYSQL, H2 -> '`' + persistedName + '`';
             case SQL_SERVER -> '[' + persistedName + ']';
-            case ORACLE ->
+            case ORACLE -> {
                 // Oracle requires quoted identifiers to be in upper case
-                '"' + persistedName.toUpperCase(Locale.ENGLISH) + '"';
+                String result = supportsDynamicValues ? SqlQueryBuilderUtils.mapPersistedName(persistedName, s -> s.toUpperCase(Locale.ENGLISH)) : persistedName.toUpperCase(Locale.ENGLISH);
```

Comment:
The root issue was if we have `@MappedEntity("${entity.prefix}_book")` then previous `quote()` implementation would turn it into `${ENTITY.PREFIX}_BOOK` and then when executing query or generating table the app cannot resolve value since in config there is `entity.prefix=tbl_common_`
So this skips uppercase for entries between `${` and `}`